src/cdk/text-field/autofill.ts

```diff
@@ -68,12 +68,18 @@ export class AutofillMonitor implements OnDestroy {
     }
 
     const result = new Subject<AutofillEvent>();
+    const cssClass = 'cdk-text-field-autofilled';
     const listener = (event: AnimationEvent) => {
-      if (event.animationName === 'cdk-text-field-autofill-start') {
-        element.classList.add('cdk-text-field-autofilled');
+      // Animation events fire on initial element render, we check for the presence of the autofill
+      // CSS class to make sure this is a real change in state, not just the initial render before
+      // we fire off events.
+      if (event.animationName === 'cdk-text-field-autofill-start' &&
+          !element.classList.contains(cssClass)) {
+        element.classList.add(cssClass);
```

Comment:
You're repeating `element.classList` in a few places here. Consider moving it out into a variable.